File: Python/DynamicProgramming/tabulationMemorization.py
def tabulationFibonacci(n: int) -> int:
    if n <=1: return n
    prev = 0 
    curr = 1
    counter = 1
    while counter < n:
        temp = curr
        curr = prev + curr
        prev = temp
        counter += 1
    return curr

print(tabulationFibonacci(10))

File: TypeScript/DynamicProgramming/tabulationMemorization.ts
const tabulationFibonacci = (n: number): number => {
    if (n <= 1) return n
    let prev = 0
    let curr = 1
    let counter = 1
    while (counter < n) {
        const temp = curr
        curr = prev + curr
        prev = temp
        counter++
    }
    return curr
}

console.log(tabulationFibonacci(10))